DMX.Core.Api/Services/Foundations/LabCommands/LabCommandService.Exceptions.cs

```diff
@@ -1,70 +1,117 @@
-                return await returningLabCommandFunction();
-            }
-            catch (InvalidLabCommandException invalidLabCommandException)
-            {
-                throw CreateAndLogValidationException(invalidLabCommandException);
-            }
-            catch (NotFoundLabCommandException notFoundLabCommandException)
+using Microsoft.EntityFrameworkCore;
+using Xeptions;
+
+namespace DMX.Core.Api.Services.Foundations.LabCommands
+{
+    public partial class LabCommandService : ILabCommandService
+    {
+        private delegate ValueTask<LabCommand> ReturningLabCommandFunction();
+
+        private async ValueTask<LabCommand> TryCatch(ReturningLabCommandFunction returningLabCommandFunction)
+        {
+            try
+            {
+                return await returningLabCommandFunction();
+            }
+            catch (NullLabCommandException nullLabCommandException)
+            {
+                throw CreateAndLogValidationException(nullLabCommandException);
+            }
+            catch (InvalidLabCommandException invalidLabCommandException)
+            {
+                throw CreateAndLogValidationException(invalidLabCommandException);
+            }
+            catch (NotFoundLabCommandException notFoundLabCommandException)
             {
                 throw CreateAndLogValidationException(notFoundLabCommandException);
-            }
-            catch (SqlException sqlException)
+            }
+            catch (SqlException sqlException)
             {
-                var failedLabCommandStorageException = new FailedLabCommandStorageException(sqlException);
+                var failedLabCommandStorageException =
+                    new FailedLabCommandStorageException(sqlException);
 
-                throw CreateAndLogDepedendencyException(failedLabCommandStorageException);
-            }
+                throw CreateAndLogCriticalDependencyException(failedLabCommandStorageException);
+            }
+            catch (DuplicateKeyException duplicateKeyException)
+            {
+                var alreadyExistsLabException =
+                    new AlreadyExistsLabCommandException(duplicateKeyException);
+
+                throw CreateAndLogDependencyValidationException(alreadyExistsLabException);
+            }
+            catch (DbUpdateException dbUpdateException)
+            {
+                var failedLabCommandStorageException =
+                    new FailedLabCommandStorageException(dbUpdateException);
+
+                throw CreateAndLogDependencyException(failedLabCommandStorageException);
+            }
             catch (Exception exception)
             {
                 var failedLabCommandServiceException = new FailedLabCommandServiceException(exception);
 
                 throw CreateAndLogServiceException(failedLabCommandServiceException);
-            }
+            }
+        }
+
+        private Xeption CreateAndLogValidationException(Xeption exception)

```

Comment:
Return type should be `LabCommandValidationException` because that's what the method is promising to return.
The input should always be `Xeption` just like you did.